src/integrationTest/cloudformation/templateRegistryManager.test.ts

```diff
@@ -19,118 +19,116 @@ import { getTestWorkspaceFolder } from '../integrationTestsUtilities'
+        assert.strictEqual(await registryHasTargetNumberOfFiles(registry, 1), true)
 
-        const initialObj = registry.getRegisteredTemplate(path1)
+        const initialObj = registry.getRegisteredTemplate(filepath)
         assert.ok(initialObj)
         if (initialObj) {
-            objectsFound++
             assert.strictEqual(Object.keys(initialObj.template).includes('Globals'), false)
         }
 
         await strToYamlFile(makeSampleSamTemplateYaml(true), filename.fsPath)
 
         await new Promise(resolve => setTimeout(resolve, 500))
-        assert.strictEqual(registry.registeredTemplates.length, 1)
+        assert.strictEqual(await registryHasTargetNumberOfFiles(registry, 1), true)
 
-        const editedObj = registry.getRegisteredTemplate(path1)
+        const editedObj = registry.getRegisteredTemplate(filepath)
         assert.ok(editedObj)
         if (editedObj) {
-            objectsFound++
             assert.strictEqual(Object.keys(editedObj.template).includes('Globals'), true)
         }
-
-        assert.strictEqual(objectsFound, 2)
     })
 
     it('can handle deleted files', async () => {
-        const path1 = path.join(testDir, 'deleteMe.yml')
-        additionalFiles.push(path1)
+        await manager.addTemplateGlob('**/deleteMe.yml')
 
-        const filename = vscode.Uri.file(path1)
-        await strToYamlFile(makeSampleSamTemplateYaml(false), filename.fsPath)
+        // Specifically creating the file after the watcher is added
+        // Otherwise, it seems the file is deleted before the file watcher realizes the file exists
+        // This way, we know that a file watcher detects the create event on this file and thus is tracking it
+        const filepath = path.join(testDir, 'deleteMe.yml')
 
-        await manager.addTemplateGlob('**/deleteMe.yml')
+        const filename = vscode.Uri.file(filepath)
+        await strToYamlFile(makeSampleSamTemplateYaml(false), filename.fsPath)
 
-        await new Promise(resolve => setTimeout(resolve, 500))
-        assert.strictEqual(registry.registeredTemplates.length, 1)
+        assert.strictEqual(await registryHasTargetNumberOfFiles(registry, 1), true)
 
-        const fileToDelete = additionalFiles.pop()
-        if (fileToDelete) {
-            await rmrf(fileToDelete)
-        }
+        await rmrf(filepath)
 
-        await new Promise(resolve => setTimeout(resolve, 500))
-        assert.strictEqual(registry.registeredTemplates.length, 0)
+        assert.strictEqual(await registryHasTargetNumberOfFiles(registry, 0), true)
     })
 })
+
+async function registryHasTargetNumberOfFiles(
+    registry: CloudFormationTemplateRegistry,
+    target: number
+): Promise<boolean> {
+    while (registry.registeredTemplates.length !== target) {
+        await new Promise(resolve => setTimeout(resolve, 20))
+    }
+
+    return true
+}
+
+async function createTestYamlTemplates(testDir: string, testDirNested: string) {
```

Comment:
This seems rather unconventional. It also obscures the difference in file extensions being written. The input paths are arbitrary. Suggest to change parameters to be an array of full paths to write files to. Each path gets the same file contents written out in this method.